### PR TITLE
unix: get mainline kernel version in Ubuntu

### PR DESCRIPTION
**Pull request by santigimeno**
In Ubuntu, the kernel version reported by `uname()` follows the versioning format that Ubuntu uses for their kernels which does not have a direct correspondence with the mainline kernel version they're based on. Get that version from `/proc/version_signature` as documented in:

https://wiki.ubuntu.com/Kernel/FAQ#Kernel.2FFAQ.2FGeneralVersionRunning.How_can_we_determine_the_version_of_the_running_kernel.3F

### CODE REVIEW

File: src/unix/linux.c [hunk 1]
```diff
@@ -323,11 +323,23 @@ unsigned uv__kernel_version(void) {
...
     return version;
 
+  /* Check /proc/version_signature first as it's the way to get the mainline
+   * kernel version in Ubuntu. The format is:
+   *   Ubuntu ubuntu_kernel_version mainline_kernel_version
+   * For example:
+   *   Ubuntu 5.15.0-79.86-generic 5.15.111
+   */
+  if (0 == uv__slurp("/proc/version_signature", v_sig, sizeof(v_sig)))
+    if (4 == sscanf(v_sig, "Ubuntu %s %u.%u.%u", ubuntu_vers, &major, &minor, &patch))
```
Comment (bnoordhuis) [inline src/unix/linux.c:340]:
```suggestion
    if (3 == sscanf(v_sig, "Ubuntu %*s %u.%u.%u", &major, &minor, &patch))
```

**Review by bnoordhuis — APPROVED** [review]
LGTM with a suggestion. Good catch.

Debian has a similar issue, the real kernel version is in `u.version` (uname -v), not in `u.release` (uname -r)

`u.version` is fairly free-form. A proper scanner looks like this:
```c
needle = strstr(u.version, "Debian ");
if (needle != NULL)
  if (3 == sscanf(needle, "Debian %u.%u.%u", &major, &minor, &patch))
    goto calculate_version;
```

**Comment by santigimeno** [comment]
> LGTM with a suggestion. Good catch.
> 
> Debian has a similar issue, the real kernel version is in `u.version` (uname -v), not in `u.release` (uname -r)
> 
> `u.version` is fairly free-form. A proper scanner looks like this:
> 
> ```c
> needle = strstr(u.version, "Debian ");
> if (needle != NULL)
>   if (3 == sscanf(needle, "Debian %u.%u.%u", &major, &minor, &patch))
>     goto calculate_version;
> ```

Are you ok if I add this code as well?

File: src/unix/linux.c (hunk 1 above)
Comment (santigimeno) [reply src/unix/linux.c:340]: Didn't know about the `*`. Thanks!

**Comment by bnoordhuis** [comment]
Of course! :)

**Comment by santigimeno** [comment]
@bnoordhuis I added the Debian check in a new commit (and set yourself as the Author). PTAL. Thanks!

**Review by bnoordhuis — APPROVED** [review]

**Pull request merged**